### ISSUE TITLE
Create form for users to register

### ISSUE THREAD

**Issue by diwanow** [RailsGirlsCPH/mentor-mentee-platform#8]
As a potential user
I can register in the system by providing a unique username and a password
So that I can use the system

**Acceptance criteria**

- There is a form with 2 fields: username and password
- There is a Submit button
- Password is not visible in plain text (user doesn't see the characters they type)
- Validation rules
    - the username must be unique for the system i.e. there is no existing user with the same username in the DB
   - validation happens when user presses Submit button
- If information entered by the user is valid, the user is created in the DB
- If information entered by the user is not valid, nothing happens

Depends on #7 to be done first

**Comment by mikker** [comment]
Well established solutions exist. Most widespread is probably [Devise](https://github.com/plataformatec/devise) but [Clearance](https://github.com/thoughtbot/clearance) is also a great option, depending on your tastes.

In the category of shameless self-promotion there's also my own [Passwordless](https://github.com/mikker/passwordless) but that doesn't fit your description perfectly as it needs no passwords (for the better!)

**Comment by diwanow** [comment]
@mikker I added a few Spikes to investigate (rather learn) the options you mentioned. Thanks for the suggestions!

**Comment by sibsmc** [comment]
I am going to close this issue as backend are creating API's not forms, however work with regards to validation of user data needs to be done, will create new issue for this.
